### ISSUE TITLE
multivariate multiple regression support

### ISSUE THREAD

**Issue by jkylearmstrong** [tidymodels/tidymodels#118]
It would appear that tidymodels / parsnip does not support multivariate multiple regression (multiple outcomes with multiple features): [https://community.rstudio.com/t/can-tidymodels-xgboost-predict-2-things-multivariate-at-once-regression/136245](https://community.rstudio.com/t/can-tidymodels-xgboost-predict-2-things-multivariate-at-once-regression/136245)

**Comment by JavOrraca** [comment]
+1 @jkylearmstrong 's feature request! I came across your [Posit Community post on this same topic](https://forum.posit.co/t/multivariate-multiple-regression-with-tidymodels/182013) and I agree multi-label modeling with tidymodels would be amazing. This is something we do today with Python's {[sklearn.multioutput](https://scikit-learn.org/stable/modules/multiclass.html#multiclass-and-multioutput-algorithms)} and we get great results when pairing with gradient boosted tree algos like [XGBoost](https://xgboost.readthedocs.io/en/stable/tutorials/multioutput.html) or [LightGBM](https://forecastegy.com/posts/lightgbm-multi-output-regression-classification-python/).

**Comment by topepo** [comment]
Subject to other constraints, we can work on this once we have completed the current development cycle (sparsity, postprocessing, and feature selection). 

We'll need to figure out methods and syntax for multivariate performance measures. I've seen that in the spectroscopy literature but I don't know of any good, general reference for doing this. Please let use know if you are aware of any. 

@JavOrraca multi-label outcomes are different in a lot of ways from multivariate outcomes. There are some of the same engineering issues but many differences. I'm not trying to nitpick, but I wanted to make sure that we're talking about the same thing. In some ways, multivariate (regression) is much easier to do (and parsnip already can do it).

**Comment by JavOrraca** [comment]
Thanks @topepo, `tidymodels` is definitely my preference over `sklearn` so I just structure my data to make it work as needed with R.
